Data/Primitive/Addr.hs

```diff
@@ -100,3 +107,11 @@ setAddr :: (Prim a, PrimMonad m) => Addr -> Int -> a -> m ()
 {-# INLINE setAddr #-}
 setAddr (Addr addr#) (I# n#) x = primitive_ (setOffAddr# addr# 0# n# x)
 
+-- | Convert an 'Addr' to an 'Int'.
+addrToInt :: Addr -> Int
+{-# INLINE addrToInt #-}
+addrToInt (Addr addr#) = I# (addr2Int# addr#)
```

Comment:
wouldn't Word be more correct / robustly portable?
(granted both options are sad and very very very ghc RTS/semantics specific, so i can't complain too much :) )